src/_DraggableBase.js

```diff
@@ -0,0 +1,443 @@
+        }
+        this.checkScroll(touchMovement);
+        this.computeMovement(touchMovement);
+      }
+    }
+  },
+  onTouchEnd: function onTouchEnd(touch = {}) {
+    if (this._source) {
+      this.placeItem(touch)
+          .removeStyling()
+          .clearValues();
+    }
+  },
+  placeItem: function placeItem() {
+    this._source = this._container.removeChild(this._source);
+    if (this._previousElement) {
+      // This accounts for when the source is between two nodes or the last element in the container
+      domConstruct.place(this._source, this._previousElement, 'after');
+    } else {
+      // This is the situation in which the source was placed as the first element of the container
+      domConstruct.place(this._source, this._nextElement, 'before');
+    }
+    domStyle.set(this._source, {
+      top: '',
+    });
+    return this;
+  },
+  removeStyling: function removeStyling() {
+    domStyle.set(this._source, {
+      opacity: '',
+      zIndex: '',
+      position: '',
+      width: '',
+    });
+    domStyle.set(this._container, {
+      overflow: '',
+    });
+    if (this._previousElement) {
+      this.resetMargins(this._previousElement, 'bottom');
+    } else {
+      this.resetMargins(this._nextElement, 'top');
+    }
+    return this;
+  },
+  resetMargins: function resetMargins(element = {}, marginType = {}) {
+    if (marginType === 'bottom') {
+      domStyle.set(element, {
+        marginBottom: element.previousMargin + 'px',
+      });
+    }else if (marginType === 'top') {
+      domStyle.set(element, {
+        marginTop: element.previousMargin + 'px',
+      });
+    }
+    return this;
+  },
+  scrollSmooth: function scrollSmooth(toScroll = {}, speed = {}) {
+    const currentScrollTop = toScroll.scrollTop;
+    if (speed < 0) {
+      if (currentScrollTop >= Math.abs(speed)) {
+        toScroll.scrollTop = currentScrollTop + speed;
+        this.computeMovement(this._scrollingTouch);
+      } else {
+        toScroll.scrollTop = 0;
+        this.clearScrollTimer();
+      }
+    } else {
+      if (toScroll.scrollHeight - currentScrollTop - toScroll.offsetHeight >= speed) {
+        toScroll.scrollTop = currentScrollTop + speed;
+        this.computeMovement(this._scrollingTouch);
+      } else {
+        toScroll.scrollTop = toScroll.scrollHeight - toScroll.offsetHeight + this._position.h;
+        this.clearScrollTimer();
+      }
+    }
+  },
+  scrollTimer: function scrollTimer() {
+    let scrollSpeed = 0;
+    if (this._scrollDirection === 'down' && this._source.offsetTop < this._container.lastChild.offsetTop + this._position.h && this._source !== this._container.lastChild) {
+      let x = (this._scrollingTouch.pageY - this._scrollerPos.offset - this._scrollerPos.h * (1 - this.scrollAt)) / (this._scrollerPos.h * this.scrollAt); // (this._scrollerPos.h - this._scrollerPos.h * (1 - this.scrollAt))
+      if (x < 0) {
+        x = 0;
+      }
+      scrollSpeed = x * x * this.scrollSpeed;
```

Comment:
should this scroll speed calculation be in a function  with parameter to slow or increase the speed?